website/src/pages/[organism]/api/sequences.ts

```diff
@@ -0,0 +1,185 @@
+        },
+    });
+};
+
+function getSearchParams(url: URL) {
+    const searchParams = url.searchParams;
+
+    const segment = searchParams.get('segment') ?? undefined;
+    const downloadFileBasename = searchParams.get('downloadFileBasename') ?? 'sequences';
+    const headerFields = searchParams.getAll('headerFields');
+    searchParams.delete('segment');
+    searchParams.delete('headerFields');
+    searchParams.delete('downloadFileBasename');
+
+    return {
+        segment,
+        headerFields,
+        downloadFileBasename,
+        queryFilters: Object.fromEntries(searchParams),
+    } satisfies SearchParams;
+}
+
+async function getAccessionVersionToFastaHeaderMap(lapisClient: LapisClient, searchParams: SearchParams) {
+    const details = await lapisClient.getDetails({
+        ...searchParams.queryFilters,
+        fields: [ACCESSION_VERSION_FIELD, ...searchParams.headerFields],
+    });
+
+    return details.map((it) => {
+        const fastaHeaderMap = new Map<string, string>();
+
+        for (const datum of it.data) {
+            const { [ACCESSION_VERSION_FIELD]: accessionVersion, ...rest } = datum;
+            const fastaHeader = Object.values(rest).join('/');
+            fastaHeaderMap.set(accessionVersion as string, fastaHeader);
+        }
+
+        return {
+            fastaHeaderMap,
+            dataVersion: it.info.dataVersion,
+        };
+    });
+}
+
+function streamFasta(
+    sequencesResponse: AxiosResponse<Readable>,
+    fastaHeaderMap: Map<string, string>,
+    searchParams: SearchParams,
+) {
+    const sequenceName = searchParams.segment ?? 'main';
+    const ndjsonLineSchema = z.object({
+        [ACCESSION_VERSION_FIELD]: z.string(),
+        [sequenceName]: z.string(),
+    });
+
+    let streamEnded = false;
+
+    return new ReadableStream({
+        start(controller) {
+            let buffer = '';
+            const encoder = new TextEncoder();
+
+            function error(reason: unknown) {
+                if (streamEnded) {
+                    return;
+                }
+                controller.error(reason);
+                streamEnded = true;
+            }
+
+            function processBuffer() {
+                if (streamEnded) {
+                    return;
+                }
+
+                const lines = buffer.split('\n');
+                buffer = lines.pop() ?? '';
+
+                for (const line of lines) {
+                    if (line.trim() === '') {
+                        continue;
+                    }
+
+                    try {
+                        const data = ndjsonLineSchema.parse(JSON.parse(line));
+
+                        const fastaHeader = fastaHeaderMap.get(data.accessionVersion as string);
+                        if (!fastaHeader) {
+                            const reason = `Did not find fasta header for accession version ${data.accessionVersion}`;
```

Comment:
```suggestion
                            const reason = `Did not find metadata for accession version ${data.accessionVersion}`;
```

Would this be clearer? Or maybe point out that the details is missing values?